src/components/Cart/PriceDetails.jsx

```diff
@@ -0,0 +1,60 @@
+import { useNavigate } from "react-router-dom";
+
+const PriceDetails = ({ totalItems, cartTotal, login }) => {
```

Comment:
Hi Rishi,

Instead of using prop drilling to get totalItems, cartTotal and login, why don't you directly set cart and islogged in a centralised context and use those context states over here?

Then you can simply use .reduce() on the state cart array and derive totalItems and cartTotal.

Reply:
Hey @shlok30 I need to change this in all places where I used these field but I noted this as time is less I will definitely do that what you are suggesting as of now can I move with this